### ISSUE TITLE
log interval

### ISSUE THREAD

**Issue by KwangryeolPark** [huggingface/transformers#26840]
### Feature request

I want you to add "log_interval" argument for logging every N iteration not a epoch.

### Motivation

Some fine-tuning task/dataset does not require much epochs such as CoLA. It only requires 2~3 epochs, which means the log data(ex metric) is produced 2~3 times.

I want more log during training with 2~3 epochs, so I hope you add some feature which logs every N iteration.

I already add the argument in my project just adding below code, but it is borthing to add every project whenever I create new project based on your codes.

`
if completed_steps % args.log_interval == 0:
# presented evaluation code in original code.
`

### Your contribution

`
if completed_steps % args.log_interval == 0:
                model.eval()
                samples_seen = 0
                for step, batch in enumerate(eval_dataloader):
                    with torch.no_grad():
                        outputs = model(**batch)
                    predictions = outputs.logits.argmax(dim=-1) if not is_regression else outputs.logits.squeeze()
                    predictions, references = accelerator.gather((predictions, batch["labels"]))
                    # If we are in a multiprocess environment, the last batch has duplicates
                    if accelerator.num_processes > 1:
                        if step == len(eval_dataloader) - 1:
                            predictions = predictions[: len(eval_dataloader.dataset) - samples_seen]
                            references = references[: len(eval_dataloader.dataset) - samples_seen]
                        else:
                            samples_seen += references.shape[0]
                    metric.add_batch(
                        predictions=predictions,
                        references=references,
                    )

                eval_metric = metric.compute()
                logger.info(f"epoch {epoch}: {eval_metric}")

                if args.with_tracking:
                    accelerator.log(
                        {
                            "accuracy" if args.task_name is not None else "glue": eval_metric,
                            "train_loss": total_loss.item() / args.log_interval,
                            "train_mem": total_memory / args.log_interval,
                            "epoch": epoch,
                            "step": completed_steps,
                            "lr": optimizer.param_groups[0]["lr"],
                        },
                        step=completed_steps,
                    )
                total_loss = 0
`

**Comment by trishanu-init** [comment]
@KwangryeolPark Please assign this issue to me
